src/linux/system.rs

```diff
@@ -302,7 +301,7 @@ impl SystemExt for System {
             users: Vec::new(),
             boot_time: boot_time(),
         };
-        if !refreshes.cpu() {
+        if refreshes.cpu() {
             s.refresh_processors(false); // We need the processors to be filled.
         }
         s.refresh_specifics(refreshes);
```

Comment:
Will this now result in two calls to `refresh_processors(false)`, one of them unnecessary? I'm thinking you can delete lines 304-306?

Reply:
It's a bit more complex than that. I'll send another PR for a more complete fix.